### ISSUE TITLE
scrollDownRecyclerView(int index) and scrollRecyclerViewToBottom(int index) do the same thing

### ISSUE THREAD

**Issue by brigitlyons** [RobotiumTech/robotium#804]
Using Robotium 5.5.3.

Both `scrollDownRecyclerView(int index)` and `scrollRecyclerViewToBottom(int index)` do the exact same thing. I would expect the behavior of `scrollRecyclerViewToBottom(int index)` to function like `scrollListToBottom(AbsListView view)`, which will scroll to the bottom of the current page and thus cause more items (if available) to load.

Instead, both drag the screen up by one screen height.

**Comment by renas** [comment]
Thanks for reporting this. This will be fixed shortly.

**Comment by renas** [comment]
In the mean time you could try to use: while (solo.scrollDownRecyclerView).

**Comment by brigitlyons** [comment]
Hello. The suggested work around is essentially how `scrollRecyclerViewToBottom()` is currently implemented (see [here](https://github.com/RobotiumTech/robotium/blob/master/robotium-solo/src/main/java/com/robotium/solo/Solo.java#L2208) and [here](https://github.com/RobotiumTech/robotium/blob/master/robotium-solo/src/main/java/com/robotium/solo/Scroller.java#L144])), so it will not work.

The problem is that `scrollDownRecyclerView()` will return false after performing one scroll.

I think this is due to using [view.getScrollY()](https://github.com/RobotiumTech/robotium/blob/master/robotium-solo/src/main/java/com/robotium/solo/Scroller.java#L130) on a RecyclerView, which will almost always zero.

**Comment by renas** [comment]
Yes, View.getScrollY() used to return non-zero before for RecyclerViews. We will implement another way of doing this. Thanks again for reporting this.

**Comment by younas-bangash** [comment]
I search alot but did not find any tutorial how to test the RecyclerViews can you please refer me link how to test the RecyclerViews I am using following code 

```
int itemCount = ((LinearLayoutManager) recyclerView.getLayoutManager ()).findLastVisibleItemPosition ()
                - ((LinearLayoutManager) recyclerView.getLayoutManager ()).findFirstVisibleItemPosition () + 1;
 final boolean res=mTestingActivity.scrollRecyclerViewToBottom(itemCount );
```
